cylc/rose/stem.py

```diff
@@ -369,7 +369,7 @@ def _generate_name(self):
                 basedir = os.path.abspath(self.opts.source)
             else:
                 basedir = os.getcwd()
-        name = os.path.basename(basedir)
+        name = os.path.basename(os.path.dirname(basedir))
```

Comment:
This is in the wrong place - it will change the name in all cases rather than just the case we're trying to fix